src/sql/workbench/browser/modelComponents/treeViewDataProvider.ts

```diff
@@ -11,6 +11,23 @@ import { IModelViewTreeViewDataProvider, ITreeComponentItem } from 'sql/workbenc
 import { INotificationService } from 'vs/platform/notification/common/notification';
 // eslint-disable-next-line code-import-patterns
 import * as vsTreeView from 'vs/workbench/api/browser/mainThreadTreeViews';
+import { ResolvableTreeItem } from 'vs/workbench/common/views';
+
+export class ResolvableTreeComponentItem extends ResolvableTreeItem implements ITreeComponentItem {
+
+	checked?: boolean;
+	enabled?: boolean;
+	onCheckedChanged?: (checked: boolean) => void;
+	children?: ITreeComponentItem[];
+
+	constructor(treeItem: ITreeComponentItem, resolve?: (() => Promise<ITreeComponentItem | undefined>)) {
+		super(treeItem, resolve);
+		this.checked = treeItem.checked;
+		this.enabled = treeItem.enabled;
+		this.onCheckedChanged = treeItem.onCheckedChanged;
+		this.children = treeItem.children;
```

Comment:
Is this safe to do?

Reply:
Hm, from what perspective? The array or the child items within it?

(honestly this seems suspect in the first place. Not sure what the use case is for storing this like it is)

Reply:
Looking at the use I don't see any immediate issues with using the children like this. When we refresh the children of a node it creates the list of ResolvableTreeComponentItems and sets that to this list so that should be ok.

I'll dig into this further but for getting this fix out since this is only used in one place that we know of and I haven't seen any issues in my testing I think it's fine to do as is. 

I need to also follow up on making sure the handlers are disposed of properly (they probably aren't)